code/freespace2/freespace.cpp

```diff
@@ -7590,11 +7586,7 @@ void game_render_mouse(float frametime)
 	if ( am->first_frame != -1 ) {
 		mouse_get_pos(&mx, &my);
 		am->elapsed_time += frametime;
-		am->current_frame = fl2i( ( am->elapsed_time / am->time ) * (am->num_frames-1) );
-		if ( am->current_frame >= am->num_frames ) {
-			am->current_frame = 0;
-			am->elapsed_time = 0.0f;
-		}
+		am->current_frame = bm_get_anim_frame(am->first_frame, am->elapsed_time, 0.0f, true);
```

Comment:
This code may behave slightly differently because `elapsed_time` isn't reset to zero. Floating point numbers are more precise close to zero which means that repeatedly adding the frametime may lead to increasingly large precision losses.